docs/docs/development/hardware-integration/dongle.mdx

```diff
@@ -0,0 +1,302 @@
+
+Navigate to the directory defining your keyboard (in-tree keyboards found [here](https://github.com/zmkfirmware/zmk/tree/main/app/boards)) and look through the [devicetree files](../../config/index.md) for nodes with `compatible = "zmk,matrix-transform";`.
+This should look something like this:
+
+```dts
+    default_transform: keymap_transform_0 {
+        compatible = "zmk,matrix-transform";
+        columns = <14>;
+        rows = <5>;
+
+        map = <
+            // Lots of RC(r,c) macros
+        >;
+    };
+```
+
+Copy this node into your `my_keyboard_dongle.overlay` file. Take care to put it under the root node (the `/ { ... };` node).
+If there are multiple matrix transformations, copy the one that matches your keyboard's layout.
+
+Make a note of the label that the transform has, it will be used later. In the example the label is `default_transform`.
+
+#### Physical layout
+
+A full physical layout is necessary to allow your dongle to be used with [ZMK Studio](../../features/studio.md).
+If your keyboard is not Studio-ready or you have no interest in using ZMK Studio with your dongle, then this section is simplified significantly.
+
+<Tabs
+    groupId="studio-ready"
+    defaultValue="studio"
+    values={[
+      { label: "Dongle with Studio", value: "studio" },
+      { label: "Dongle without Studio", value: "nostudio" }
+    ]}
+  >
+<TabItem value="studio">
+
+You will need to find your keyboard's physical layout, much like finding the matrix transform.
+Depending on your keyboard, the physical layout could be found in a variety of locations. Look for a node with `compatible = "zmk,physical-layout";`.
+There are three commonly found possibilities:
+
+- The physical layout is found in a file called `my_keyboard-layouts.dtsi`
+- The physical layout is found in the same file as the matrix transform and chosen node
+- The physical layout is imported from ZMK's shared layouts - `#include <layouts/<layout_name>.dtsi>` can be found at the top of one of the devicetree files.
+
+<Tabs
+    groupId="layout-locations"
+    defaultValue="layouts"
+    values={[
+      { label: "Layouts File", value: "layouts" },
+      { label: "Base File", value: "base" },
+      { label: "Imported", value: "imported" }
+    ]}
+  >
+<TabItem value="layouts">
+Copy the file into your dongle's folder. Then import the file, assign the matrix transform to it, and select it in the `chosen` node:
+
+```dts
+#import "my_keyboard-layouts.dtsi"
+
+&physical_layout0 {
+    transform = <&default_transform>;
+};
+
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+};
+```
+
+Make sure that the labels `physical_layout0` and `default_transform` match those of the physical layout node defined in the file and the matrix transform respectively.
+
+If there are multiple physical layouts in the file, you will need to copy over all of the remaining matrix transformations and assign them similarly.
```

Comment:
Wouldn't it be better if we put the entire section related to the case of having multiple layouts under a new title?

Something like `#### If there are multiple physical layouts in the file`.

This is because the next topics seem to have no clear connection with the previous one

Look at:
> Copy the physical layout node into your...

Seems like there should be a "In this case," before the text to maintain cohesion.

Or it could be done by a new title that contains all that block.

Without it, I feel that users can get confused to know how far they should follow the instructions